src/metatrain/experimental/nanopet/modules/samples.py

```diff
@@ -0,0 +1,535 @@
+                get_edge_sample_values(systems, include_atom_types=True)
+            ),
+        )
+        # store the samples labels for the last layer PET features for per_pair
+        # (symmetrized) targets, i.e. the stacked node and edge samples
+        ll_per_pair_samples_sym = Labels(
+            edge_samples_sym.names,
+            torch.vstack(
+                [
+                    get_node_sample_values(
+                        systems,
+                        sample_kind="per_pair_sym",
+                        include_atom_type=True,
+                    ),
+                    edge_samples_sym.values,
+                ]
+            ),
+        )
+
+    # if there are spherical targets on an atomic basis, we need to store the sample
+    # labels for each basis block, used to slice the last layer PET node/edge features
+    # just before application of the output layer
+    atomic_basis_samples: Dict[str, Dict[str, Labels]] = {
+        "per_atom": {
+            "": Labels(["_"], torch.empty((0, 1), dtype=node_samples.values.dtype))
+        },
+        "per_pair": {
+            "": Labels(["_"], torch.empty((0, 1), dtype=node_samples.values.dtype))
+        },
+        "per_pair_sym": {
+            "": Labels(["_"], torch.empty((0, 1), dtype=node_samples.values.dtype))
+        },
+    }
+    if any_per_atom:
+        atomic_basis_samples["per_atom"] = samples_for_atomic_basis_per_atom(
+            systems, node_samples, atomic_types
+        )
+    if any_per_pair:
+        atomic_basis_samples["per_pair"] = samples_for_atomic_basis_per_pair(
+            systems, ll_per_pair_samples, atomic_types, sample_kind="per_pair"
+        )
+    if any_per_pair_sym:
+        atomic_basis_samples["per_pair_sym"] = samples_for_atomic_basis_per_pair(
+            systems, ll_per_pair_samples_sym, atomic_types, sample_kind="per_pair_sym"
+        )
+
+    return (
+        node_samples,
+        edge_samples,
+        edge_samples_sym,
+        ll_per_pair_samples,
+        ll_per_pair_samples_sym,
+        atomic_basis_samples,
+    )
+
+
+# ===== Samples labels for the PET node/edge features =====
+
+
+def get_node_sample_values(
+    systems: List[System],
+    sample_kind: str,
+    include_atom_type: bool,
+) -> torch.Tensor:
+    """
+    Returns a torch tensor of the sample values that correspond to the internal PET node
+    features.
+
+    If ``sample_kind="per_atom"``, the samples values are returned with 2 dimensions
+    corresponding to "system" and "atom".
+
+    If ``sample_kind="per_pair"``, the dimensions returned correspond to "system",
+    "first_atom", "second_atom", "cell_shift_a", "cell_shift_b", "cell_shift_c". As
+    these are nodes, the atom indices are equal and the cell shifts are zero.
+
+    If ``include_atom_type=True``, the atom types are prepended dimensions, either
+    corresponding to "center_type" if ``n_center=1`` or ["first_atom_type",
+    "second_atom_type"] if ``n_center=2``.
```

Comment:
There is no longer an `n_center` parameter

Reply:
Good catch